src/impl/tpls/KokkosSparse_spmv_tpl_spec_decl.hpp

```diff
@@ -70,6 +71,84 @@ struct SPMV<class AT, class AO, class AD, class AM, class AS,
     const YVector& y) {
   Kokkos::Profiling::pushRegion("KokkosSparse::spmv[TPL_CUSPARSE,double]");
   printf("Using CUSPARSE for spmv!\n");
+
+  cudaError_t      cuError;
+  cusparseStatus_t status;
+  cusparseHandle_t handle=0;
+  // cusparseMatDescr_t descr=0;
+
+  /* initialize cusparse library */
+  status = cusparseCreate(&handle);
+  if (status != CUSPARSE_STATUS_SUCCESS) {
+    throw("KokkosSparse::spmv[TPL_CUSPARSE,double]: cusparse was not initialized correctly");
+  }
+
+  /* create matrix */
+  cusparseSpMatDescr_t A_cusparse;
+  status = cusparseCreateCsr(&A_cusparse, A.numRows(), A.numCols(), A.nnz(),
```

Comment:
You're exposing cuSPARSE functions in the header file :(

Reply:
I am actually thinking of creating a small utility function that takes in a `Kokkos::CrsMatrix` and returns a `cusparseSpMatDescr_t` so part of this code will disappear down the road...
I will also look into other TPL specialization we have in kokkos-kernels and see how things are implemented. Depending on that I might move things around accordingly.

Reply:
@lucbv Check out how I hid the TPL header file includes for cuSOLVER in `tpetra/tsqr/src`.

Reply:
@mhoemmen as discussed I will outline a strategy to reduce the size of the macros in the `KokkosSparse_spmv_tpl_spec_decl.hpp` and hide the header in an implementation file.